Framework/PythonInterface/plugins/algorithms/VesuvioPeakPrediction.py

```diff
@@ -0,0 +1,162 @@
+from mantid.api import *
+from mantid.kernel import *
+from vesuvio.base import VesuvioBase
+from mantid.simpleapi import *
+
+import math
+import numpy as np
+
+
+class VesuvioPeakPrediction(VesuvioBase):
+    _model = None
+    _temperature = None
+    _atomic_mass = None
+    _frequency = None
+    _debye_temp = None
+
+    def summary(self):
+        return "Predicts parameters for Vesuvio peak widths using Debye or Einstein methods"
+
+    def category(self):
+        return 'Inelastic\\Indirect\\Vesuvio'
+
+    def PyInit(self):
+
+        self.declareProperty(name='Model', defaultValue='Einstein',
+                             validator=StringListValidator(['Debye', 'Einstein']),
+                             doc='Model used to make predictions')
+
+        arrvalid = FloatArrayBoundedValidator()
+        arrvalid.setLower(0.0)
+
+        self.declareProperty(FloatArrayProperty(name='Temperature', validator=arrvalid),
+                             doc='Temperature (K)')
+
+        floatvalid = FloatBoundedValidator(0.0)
+        floatvalid.setLowerExclusive(True)
+
+        self.declareProperty(name='AtomicMass', defaultValue=1.0,
+                             validator=floatvalid,
+                             doc='Atomic Mass (AMU)')
+
+        self.declareProperty(name='Frequency', defaultValue=1.0,
+                             validator=floatvalid,
+                             doc='Fundamental frequency of oscillator (mEV)')
+
+        self.declareProperty(name='DebyeTemperature', defaultValue=1.0,
+                             validator=floatvalid,
+                             doc='Debye Temperature (K)')
+
+    def setup(self):
+
+        self._model = self.getPropertyValue('Model')
+        self._temperature = self.getProperty('Temperature').value
+        self._atomic_mass = self.getProperty('AtomicMass').value
+        self._frequency = self.getProperty('Frequency').value
+        self._debye_temp = self.getProperty('DebyeTemperature').value
+
+    def PyExec(self):
+
+        self.setup()
+
+        vesuvio_params = CreateEmptyTableWorkspace(OutputWorkspace='vesuvio_' + self._model.lower() + '_params')
+        vesuvio_params.setTitle('Vesuvio Peak Parameters')
+        vesuvio_params.addColumn('float', 'Temperature(K)')
+        vesuvio_params.addColumn('float', 'Atomic Mass(AMU)')
+
+        if self._model == 'Einstein':
+            vesuvio_params.addColumn('float', 'Frequency(mEV)')
+            vesuvio_params.addColumn('float', 'Kinetic Energy(mEV)')
+            vesuvio_params.addColumn('float', 'Effective Temp(K)')
+            vesuvio_params.addColumn('float', 'RMS Momentum(A)')
+
+            for temp in self._temperature:
+
+                if temp == 0:
+                    temp = 1e-6
+
+                # Convert to mEV
+                temp_mev = temp / 11.604
```

Comment:
There are a few constants here that deserve being named at module level. Their value should be declared once.

Scipy has a lot of constants, for example for converting between Kelvin and meV we can have
```python
K_IN_MEV = 1e3/scipy.constants.value('Boltzmann constant in eV/K')
```
defined as a module-level constant and then

```python
temp_mev = temp/K_IN_MEV
```

and

```python
t_star = 2 * kinetic_energy * K_IN_MEV
```

Similarly other converting constants please place them at module level so that they are defined once and with a variable that describes what it does.